js/list.js

```diff
@@ -0,0 +1,204 @@
+            //console.log("value" + value);
+            if (value == "date") { //sort by date
+                arr.sort(function (a, b) {
+                    return new Date(b.date) - new Date(a.date); //convert date strings to date objects and then subtract them
+                });
+            }
+            else { //sort by name
+                arr.sort(function (a, b) {
+                    var textA = a.name.toUpperCase();
+                    var textB = b.name.toUpperCase();
+                    return (textA < textB) ? -1 : (textA > textB) ? 1 : 0;
+                });
+            }
+            console.log("Array after sorting: ");
+            console.log(arr);
+
+            let i = 0;
+            while (i < arr.length) { //load the cards
+                var name = arr[i].name;
+                //name = name.charAt(0).toUpperCase() + name.substr(1).toLowerCase();//convert to title case
+                var description = arr[i].description;
+                var date = arr[i].date;
+                console.log(name);
+                console.log(description);
+                console.log(date);
+                var newDiv = document.createElement("div");
+                var nameTextNode = document.createTextNode(name);
+                var lineBreak1 = document.createElement("br");
+                var descriptionTextNode = document.createTextNode(description);
+                var lineBreak2 = document.createElement("br");
+                var dateTextNode = document.createTextNode(date);
+
+                //Past date:red - Current date:blue - Future date:Purple  
+                newDiv.setAttribute('class', 'card');
+                if (compareDates(date) == 1)
+                    newDiv.style.backgroundColor = "#982176"; //purple
+                else if (compareDates(date) == 0)
+                    newDiv.style.backgroundColor = "#1434A4"; //blue
+                else if (compareDates(date) == -1)
+                    newDiv.style.backgroundColor = "#D22B2B"; //red
+
+                newDiv.appendChild(nameTextNode);
+                newDiv.appendChild(lineBreak1);
+                newDiv.appendChild(descriptionTextNode);
+                newDiv.appendChild(lineBreak2);
+                newDiv.appendChild(dateTextNode);
+                document.getElementById("content").appendChild(newDiv);
+                i++;
+            }
+
+            //load the count value and write it on the page
+            totalRecord = arr.length;
+            console.log("Total Record : " + totalRecord);
+            document.getElementById("count-label").innerHTML = "Count: " + totalRecord;
+
+            setTimeout(function () {
+                //check for edits
+                const allCardsArray = document.getElementsByClassName("card");
+                for (var i = 0; i < allCardsArray.length; i++) {
+                    let card = allCardsArray[i];
+                    card.addEventListener('click', function () {
```

Comment:
can we addEventListener when creating the card?

Reply:
done